### PR TITLE
DLP ORC support: part1

### PR DESCRIPTION
**Pull request by chitara-01**
<h4> Summary (Short summary of what is being done) : </h4>
Support for inspection and de-identification of ORC data stored in GCS buckets
<br>
<h4> Description (Describe in detail the fix made) : </h4>
The implementation is to read ORC files from GCS storage buckets to process the data using inspection and de-identification pipelines. The results are written in BugQuery tables. This is part one of the DLP ORC support project. Future work includes writing results as ORC files in GCS buckets and re-identify the data stored in de-identified ORC files. 
<br>
<h4> Bug ID (if any) : </h4>
301563096
<br>
<h4> Public Documentation (if any) : </h4>
<br>
<h4> TESTED (Test Cases with scenario and description - must have 1 positive and 1 negative scenario) : </h4>
1. Wrote the script to convert CSV data from CCRecords sample files to ORC format. The implementation worked successfully for inspection and de-identification.
<br>
2. Created unit tests for the implementation.
<br>
3. CI tests: to be added.

### CODE REVIEW

File: src/main/java/com/google/swarm/tokenization/orc/ORCReaderDoFn.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    public void processElement(ProcessContext context)
+            throws IOException {
+        String fileName = context.element().getKey();
+        FileIO.ReadableFile readableFile = context.element().getValue();
+        String filePath = readableFile.getMetadata().resourceId().toString();
+
+        Reader reader = new ORCFileReader().createORCFileReader(filePath, projectId);
+
+        RecordReader rows = reader.rows();
+        batch = reader.getSchema().createRowBatch();
```
Comment (dup05) [inline src/main/java/com/google/swarm/tokenization/orc/ORCReaderDoFn.java:122]: Can we add some code comments here to explain this DoFn ?

**Review by dup05 — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -93,13 +93,18 @@ dependencies {
     implementation group: 'org.apache.beam', name: 'beam-sdks-java-io-parquet', version: '2.14.0'
     implementation group: 'org.slf4j', name: 'slf4j-jdk14', version: '2.0.6'
     implementation 'com.google.cloud:google-cloud-kms:2.15.0'
-    implementation 'com.google.guava:guava:31.1-jre'
+    implementation 'com.google.guava:guava:32.1.2-jre'
```
Comment (Goutam1511) [inline build.gradle:96]: Any particular reason to upgrade guava or just routine upgrade?

File: src/main/java/com/google/swarm/tokenization/DLPTextToBigQueryStreamingV2.java [hunk 3]
```diff
@@ -218,9 +220,17 @@ private static void runInspectAndDeidPipeline(
                         .withSideInputs(headers));
         break;
       case PARQUET:
+//        TODO: Remove KeyRange parameter, as it is unused
```
Comment (saurabhgadia4) [inline src/main/java/com/google/swarm/tokenization/DLPTextToBigQueryStreamingV2.java:223]: Nit. Can you format the todo to align properly with below code?

File: src/test/java/com/google/swarm/tokenization/orc/ORCUtil.java
Comment (saurabhgadia4) [inline src/test/java/com/google/swarm/tokenization/orc/ORCUtil.java:1]: Rename the file to ORCTestUtil.java as it contains utility methods used by tests.

**Review by saurabhgadia4 — APPROVED** [review]
LGTM with minor comments.

File: src/main/java/com/google/swarm/tokenization/DLPTextToBigQueryStreamingV2.java (hunk 3 above)
Comment (chitara-01) [reply src/main/java/com/google/swarm/tokenization/DLPTextToBigQueryStreamingV2.java:223]: Noted! will update in the next PR.

**Pull request merged**